src/message-queues.ts

```diff
@@ -61,6 +61,11 @@ export interface BatchOptions {
   maxMilliseconds?: number;
 }
 
+// This is the maximum number of bytes we will send for a batch of
+// ack/modack messages. The server itself has a maximum of 512KiB, so
+// we just pull back a little from that in case of unknown fenceposts.
+export const MAX_BATCH_BYTES = 510 * 1024 * 1024;
```

Comment:
i hate to say it, but this is 534 Mib
@feywind

Reply:
confirmed this is 510 * 1024 * 1024 bytes == 534 MB = ~510 MiB so this is correct

Reply:
Uhh... yes. That slipped past both of us somehow. I'll open a new PR.